diffpy/snmf/tests/test_subroutines.py

```diff
@@ -144,8 +144,26 @@ def test_reconstruct_data(trd):
        (([[[1.5, 2], [10.5, 1], [0.5, 2]], 1]), ([[2, 2.5], [11, 1.5], [1, 2.5]])),
        (([[[-10, -10.5], [-12.2, -12.2], [0, 0]], 1]), ([[2.2, 1.7], [0, 0], [12.2, 12.2]])),
        ]
+
+
 @pytest.mark.parametrize('tld', tld)
 def test_lift_data(tld):
     actual = lift_data(tld[0][0], tld[0][1])
     expected = tld[1]
     np.testing.assert_allclose(actual, expected)
+
+tcc = [(2, [0, .5, 1, 1.5], 3, 3),
+       (3,[0,10,20,30],10,15),
+       (0,[0],11,30),
+       (5,[1,1,1,1,1,1],10000,40000),
+       (3,np.arange(stop=125,step=.05),20,2500),
+       ]
+@pytest.mark.parametrize('tcc', tcc)
+def test_create_components(tcc):
+    actual = create_components(tcc[0], tcc[1], tcc[2], tcc[3])
+    assert len(actual) == tcc[0]
+    for c in actual:
```

Comment:
I think the first and last section is just testing the constructor of the ComponentSignals container.  In other words, if the constructor is working, this will pass, and if not it won't.    The other two are just testing the __set__ method in there.  So I suggest to move these to the tests for the ComponentSignal and leaave them out here.  You could also add some tests there for edge cases if you can think of any.

There is honeslty not much to test here, so line 164 may be enough.